sled-agent/src/services.rs

```diff
@@ -549,14 +549,16 @@ impl ServiceManager {
     }
 
     // Check the services intended to run in the zone to determine whether any
-    // physical links or vnics need to be mapped into the zone when it is created.
-    //
-    // NOTE: This function is implemented to return the first link found, under
-    // the assumption that "at most one" would be necessary.
+    // physical links or vnics need to be mapped into the zone when it is
+    // created. Returns a list of links, plus whether or not they need link
+    // local addresses in the zone.
     fn links_needed(
         &self,
         req: &ServiceZoneRequest,
-    ) -> Result<Vec<Link>, Error> {
+    ) -> Result<Vec<(Link, bool)>, Error> {
+        let mut links: Vec<(Link, bool)> =
+            Vec::with_capacity(req.services.len());
```

Comment:
Tiny nit - I'd probably just make this `Vec::new()` because (a) it looks like `Nexus`/`Ntp`/`ExternalDns` may not need a link at all in the future and (b) `Maghemite` looks like it needs multiple links.

Reply:
done in b8a8fd84